google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/BuiltinMetricsTracerTest.java

```diff
@@ -0,0 +1,341 @@
+  private ArgumentCaptor<String> cluster;
+
+  private Stopwatch serverRetryDelayStopwatch;
+  private AtomicLong serverTotalRetryDelay;
+
+  @Before
+  public void setUp() throws Exception {
+    mockService = new FakeService();
+
+    serverRetryDelayStopwatch = Stopwatch.createUnstarted();
+    serverTotalRetryDelay = new AtomicLong(0);
+
+    // Add an interceptor to send location information in the trailers and add server-timing in
+    // headers
+    ServerInterceptor trailersInterceptor =
+        new ServerInterceptor() {
+          private AtomicInteger count = new AtomicInteger(0);
+
+          @Override
+          public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+              ServerCall<ReqT, RespT> serverCall,
+              Metadata metadata,
+              ServerCallHandler<ReqT, RespT> serverCallHandler) {
+            return serverCallHandler.startCall(
+                new ForwardingServerCall.SimpleForwardingServerCall<ReqT, RespT>(serverCall) {
+                  @Override
+                  public void sendHeaders(Metadata headers) {
+                    headers.put(
+                        Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER),
+                        String.format("gfet4t7; dur=%d", FAKE_SERVER_TIMING));
+                    super.sendHeaders(headers);
+                  }
+
+                  @Override
+                  public void close(Status status, Metadata trailers) {
+                    super.close(status, trailers);
+                  }
+                },
+                metadata);
+          }
+        };
+
+    serviceHelper = new FakeServiceHelper(trailersInterceptor, mockService);
+
+    statsRecorderWrapper = Mockito.mock(StatsRecorderWrapper.class);
+
+    BigtableDataSettings settings =
+        BigtableDataSettings.newBuilderForEmulator(serviceHelper.getPort())
+            .setProjectId(PROJECT_ID)
+            .setInstanceId(INSTANCE_ID)
+            .setAppProfileId(APP_PROFILE_ID)
+            .build();
+    EnhancedBigtableStubSettings.Builder stubSettingsBuilder =
+        settings.getStubSettings().toBuilder();
+    stubSettingsBuilder
+        .mutateRowSettings()
+        .retrySettings()
+        .setInitialRetryDelay(Duration.ofMillis(200));
+    stubSettingsBuilder.setTracerFactory(
+        BuiltinMetricsTracerFactory.create(
+            StatsWrapper.get(), ImmutableMap.of(), statsRecorderWrapper));
+
+    EnhancedBigtableStubSettings stubSettings = stubSettingsBuilder.build();
+    stub = new EnhancedBigtableStub(stubSettings, ClientContext.create(stubSettings));
+
+    serviceHelper.start();
+
+    longValue = ArgumentCaptor.forClass(long.class);
+    intValue = ArgumentCaptor.forClass(int.class);
+    status = ArgumentCaptor.forClass(String.class);
+    tableId = ArgumentCaptor.forClass(String.class);
+    zone = ArgumentCaptor.forClass(String.class);
+    cluster = ArgumentCaptor.forClass(String.class);
```

Comment:
I think its cleaner to annotate the member variables with `@Captor` and use the mockito rule to construct em